packages/drivers/routerlicious-driver/src/wholeSummaryDocumentStorageService.ts

```diff
@@ -233,4 +253,24 @@ export class WholeSummaryDocumentStorageService implements IDocumentStorageServi
         });
         await Promise.all(blobCachePutPs);
     }
+
+    private convertSnapshotAndBlobsToSummaryTree(
+        snapshot: ISnapshotTree,
+        blobs: Map<string, ArrayBuffer>,
+    ): ISummaryTree {
+        const builder = new SummaryTreeAssembler();
+        for (const [path, id] of Object.entries(snapshot.blobs)) {
+            const blob = blobs.get(id);
+            assert(blob !== undefined, "Cannot find blob for a given id");
+            builder.addBlob(path, IsoBuffer.from(blob).toString("utf-8"));
+        }
+        for (const [key, tree] of Object.entries(snapshot.trees)) {
+            const subtree = this.convertSnapshotAndBlobsToSummaryTree(tree, blobs);
+            builder.addTree(key, subtree);
+        }
+        return {
+            ...builder.summary,
+            unreferenced: snapshot.unreferenced,
```

Comment:
nit: You could pass `snapshot.unrferenced` to SummaryTreeAssembler's constructor and then it could add it to tree that is returned.

Reply:
Good point - added optional props here.